11-Backbonejs/Backbone-Views.md

```diff
@@ -0,0 +1,321 @@
+
+# Intro to Backbone Views
+
+## Learning Goals
+By the end of this lesson you should be able to:
+
+- Explain what Backbone views are and why they're useful
+- Create a view around a Backbone Model
+- Render HTML using a view & an Underscore Template
+- Create a view for a Backbone Collection
+- Render a collection with a view
+
+## What is a View?
+Backbone views are kind of middle-managers in the Backbone world, filling a similar role to controllers in Rails. A view's job is to coordinate between the data and the DOM. When a DOM event happens, it's the view's job to handle it and update the data as needed, and when the data changes it's the view's job to modify the DOM to match.
+
+You may have noticed in the, previous lectures, we have a jumble (technical term) of functions handling events, and drawing the tasklist to the browser etc.  This could quickly become a mess.  It's time for some structure!
+
+We will first create two views, one for a single task item and a second for the collection of Tasks.  Each view will handle drawing and event handling for it's specific domain, and the view for the collection will utilize the smaller TaskView.  
+
+![Backbone Views](images/backbone-views.png)
+
+You can see a deployed version of this app [here:](http://cheezitman.github.io/tasklist-js)
+
+## Creating A TaskView
+
+![Application Overview](images/Ada-TaskList-JS.png)
+
+A TaskView will handle drawing an individual task item and responding to events that concern just that task.  To start we create a file in `app/views/task_view.js`  Each TaskView will only be concerned with rendering a specific task and dealing with events on that task.  We've indicated it in red in the image above.  
+
+
+```JavaScript
+import Backbone from 'backbone';
+import _ from 'underscore';
+import $ from 'jquery';
+import Task from '../models/task.js';
+
+var TaskView = Backbone.View.extend({
+  initialize: function(params) {
```

Comment:
I was a little torn because `initialize` only does one thing and we've covered `initialize` in models.  So I'm not sure it's worth breaking out the functions 1 at a time.